RadixWallet/Clients/FactorSourcesClient/FactorSourcesClient+Interface.swift

```diff
@@ -295,8 +295,48 @@ extension FactorSourcesClient {
 	}
 }
 
+extension FactorSourcesClient {
+	enum ShieldFactorStatus {
+		/// A shield can be built from the available Factor Sources
+		case valid
+
+		/// At least one hardware Factor Source must be added in order to build a Shield.
+		/// Note: this doesn't mean that after adding a hardware Factor Source we would have `valid` status.
+		case hardwareRequired
+
+		/// One more Factor Source, of any kind, must be added in order to build a Shield.
+		case anyRequired
+	}
+
+	// TODO: Move to Sargon? Maybe we can get it from Alex's work
+	func getShieldFactorStatus() async throws -> ShieldFactorStatus {
+		let factorSources = try await getFactorSources()
+		let totalCount = factorSources.count
+		let hardwareCount = factorSources.filter(\.kind.isHardware).count
+		if hardwareCount < 1 {
+			return .hardwareRequired
+		} else if totalCount < 2 {
```

Comment:
According to designs and [Confluence page](https://radixdlt.atlassian.net/wiki/spaces/AT/pages/3758063620/MFA+Rules+for+Factors+and+Security+Shields#Factor-Prerequisites) we should not count in the Biometrics/Device factor source.